numpy/lib/tests/test_index_tricks.py

```diff
@@ -169,6 +169,34 @@ def test_simple_1(self):
         assert_equal(a[:, :3, [1, 2]], a[s_[:, :3, [1, 2]]])
 
 
+class TestIx_(TestCase):
+    def test_regression_1(self):
+        # Test empty inputs create ouputs of indexing type, gh-5804
+        # Test both lists and arrays
+        for func in (range, np.arange):
```

Comment:
Wait, why is it desirable that `np.arange(0, dtype=float)` is converted to `dtype=intp`? If the user is explicit about what they asked for, it seems wrong to override them